bundles/org.openhab.binding.ecovacs/src/main/feature/feature.xml

```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<features name="org.openhab.binding.ecovacs-${project.version}" xmlns="http://karaf.apache.org/xmlns/features/v1.4.0">
+	<repository>mvn:org.openhab.core.features.karaf/org.openhab.core.features.karaf.openhab-core/${ohc.version}/xml/features</repository>
+
+	<feature name="openhab-binding-ecovacs" description="Ecovacs Binding" version="${project.version}">
+		<feature>openhab-runtime-base</feature>
+		<feature dependency="true">openhab.tp-hivemqclient</feature>
+		<bundle dependency="true">mvn:org.igniterealtime.smack/smack-tcp/4.3.3</bundle>
+		<bundle dependency="true">mvn:org.jxmpp/jxmpp-core/0.6.3</bundle>
+		<bundle dependency="true">mvn:org.jxmpp/jxmpp-jid/0.6.3</bundle>
+		<bundle dependency="true">mvn:org.jxmpp/jxmpp-util-cache/0.6.3</bundle>
+		<bundle dependency="true">mvn:org.minidns/minidns-core/0.3.3</bundle>
+		<bundle dependency="true">mvn:org.igniterealtime.smack/smack-core/4.3.3</bundle>
+		<bundle dependency="true">mvn:org.igniterealtime.smack/smack-im/4.3.3</bundle>
+		<bundle dependency="true">mvn:org.igniterealtime.smack/smack-extensions/4.3.3</bundle>
+		<bundle dependency="true">mvn:org.igniterealtime.smack/smack-sasl-javax/4.3.3</bundle>
```

Comment:
Despite listing this dependency, it looks like it is not installed automatically when installing a kar file built from these sources (see [here](https://community.openhab.org/t/ecovacs-vacuum-cleaners-binding/132989/18)). I'd appreciate any guidance from somebody more experienced with OSGi than me about what I'm missing here.
(@wborn or @cweitkamp perhaps? Sorry for the somewhat random tagging, I just selected you because I found you being involved in previous discussions about related topics. If I mistagged you, it would be great if you could hint me at who the right person to ask is ... thanks!)

Reply:
At first sight, I see no reason why they wouldn't get downloaded automatically. What makes you say they aren't?

Reply:
The forum feedback suggests it isn't installed automatically:
- https://community.openhab.org/t/ecovacs-vacuum-cleaners-binding/132989/18 (this is the one I linked to above)
- https://community.openhab.org/t/ecovacs-vacuum-cleaners-binding/132989/33 (second user with the same symptoms)

Reply:
I guess the issue is that classes from this bundle are loaded dynamically, with no import statement in the manifest of the loading bundle. `dependency="true"` essentially means "install if it is required by some other bundle" and this requirement is determined by the other bundle's import statements. You can try do remove the dependency attribute (like the other two bundles at the end of the feature).

Reply:
Thanks, I'll give that a try :+1: 
I assume I should do the same to the other smack bundles as well them, right?
(FWIW, I copied those lines from the [xmppclient binding](https://github.com/openhab/openhab-addons/blob/main/bundles/org.openhab.binding.xmppclient/src/main/feature/feature.xml#L8-L23) - I assume it has the same basic issue, but it went unnoticed because the servers people use that binding with don't use SASL?)

Reply:
No, just for the bundles that are not installed automatically. This is a work-around to force installation, in general it is better to leave the dependency installation to the framework. Maybe the binding itself uses a class from that bundle.

Reply:
That worked, thanks :+1: